File: frontend/src/dto/event-group-dto.ts
export interface EventGroupDTO {
    id: number
    fingerprint: string
    message: string
    source: string
    environment: string
    severity: string
    count: number
    firstSeen: string
    lastSeen: string
    muted: boolean
    tags: string | null
    exceptionClass: string | null
    stacktrace: string | null
    sourceFile: string | null
    sourceLine: number | null
    sourceMethod: string | null
    linkedIssueId: number | null
    projectId: number
}

File: backend/src/main/java/io/clamped/cloud/event/EventGroupDto.java
package io.clamped.cloud.event;

import io.clamped.cloud.issue.Severity;

import java.time.Instant;

public record EventGroupDto(
        Long id,
        String fingerprint,
        String message,
        EventSource source,
        EventEnvironment environment,
        Severity severity,
        Long count,
        Instant firstSeen,
        Instant lastSeen,
        boolean muted,
        String tags,
        String exceptionClass,
        String stacktrace,
        String sourceFile,
        Integer sourceLine,
        String sourceMethod,
        Long linkedIssueId,
        Long projectId
) {}
